### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by VoidScute** [rtckit/php-sip#20] [Q&A] [answered]
Hello,

I m newbie with sip and have a question. 
I read the readme and think the anwser to my question is yes but i have to be sure. 
- Can i make a phone ringing and display (on the phone) a custom message.

For what i understand the documentation part for my use case is "SIP Message Rendering" and i guess "SIP Message Parsing" to be sure everything run smothly (aka : following protocol) ? 

Thx for your time

**Comment by cdosoftei** [comment, marked as answer]
[php-sip](https://github.com/rtckit/php-sip) is a SIP message parsing/rendering library, so its purpose is to decode raw SIP messages (plain text) into structured objects and vice-versa.

While parsing/rendering is definitely required in order to place and receive calls, there's more to it, at the very least handling the networking I/O and the messages within a SIP dialog; this is outside of the scope of this library and effectively describes a [SIP user agent](https://www.rfc-editor.org/rfc/rfc3261#section-8).

You might wish to check [eqivo](https://github.com/rtckit/eqivo), it's an API layer wrapped around [FreeSWITCH](https://github.com/signalwire/freeswitch) which might streamline your telephony integration.
